### PR TITLE
Allow to build container behind a proxy

### PR DESCRIPTION
**Pull request by chris060986**
Allows to specifiy a proxy to build the container behind a proxy like cntlm.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -98,6 +98,8 @@ bin/$(OS)_$(ARCH)/$(BIN): build-dirs
 	    -v $$(pwd)/bin/$(OS)_$(ARCH):/go/bin/$(OS)_$(ARCH)                       \
 	    -v $$(pwd)/.go/std/$(OS)_$(ARCH):/usr/local/go/pkg/$(OS)_$(ARCH)_static  \
 	    -v $$(pwd)/.go/cache:/.cache                                             \
+            --env HTTP_PROXY=$(HTTP_PROXY)                                 	     \
```
Comment (thockin) [inline Makefile]: indenting is messed up.  Please use tabs to the general indent level, then spaces.  In this case one tab to align with the command (@docker) and then 4 spaces.  Also use spaces as trailing whitespace.

What I see in my editor:


```
>-------@docker run                                                                  \
>-------    -i                                                                       \
>-------    -u $$(id -u):$$(id -g)                                                   \
>-------    -v $$(pwd)/.go:/go                                                       \
>-------    -v $$(pwd):/go/src/$(PKG)                                                \
>-------    -v $$(pwd)/bin/$(OS)_$(ARCH):/go/bin                                     \
>-------    -v $$(pwd)/bin/$(OS)_$(ARCH):/go/bin/$(OS)_$(ARCH)                       \
>-------    -v $$(pwd)/.go/std/$(OS)_$(ARCH):/usr/local/go/pkg/$(OS)_$(ARCH)_static  \
>-------    -v $$(pwd)/.go/cache:/.cache                                             \
            --env HTTP_PROXY=$(HTTP_PROXY)                                 >----     \                                                                                     
            --env HTTPS_PROXY=$(HTTPS_PROXY)                                         \
>-------    -w /go/src/$(PKG)                                                        \
>-------    --rm                                                                     \
>-------    $(BUILD_IMAGE)                                                           \
>-------    /bin/sh -c "                                                             \
>-------        ARCH=$(ARCH)                                                         \
>-------        OS=$(OS)                                                             \
>-------        VERSION=$(VERSION)                                                   \
>-------        PKG=$(PKG)                                                           \
>-------        ./build/build.sh                                                     \
>-------    "
```

Comment (thockin) [reply Makefile]: BTW: `>-------` is a tab :)

File: Makefile [hunk 2]
```diff
@@ -110,7 +112,6 @@ bin/$(OS)_$(ARCH)/$(BIN): build-dirs
 	    "
 
 DOTFILE_IMAGE = $(subst /,_,$(IMAGE))-$(TAG)
-
```
Comment (thockin) [inline Makefile]: leave this line please

File: Makefile [hunk 3]
```diff
@@ -119,7 +120,7 @@ container: .container-$(DOTFILE_IMAGE) container-name
 	    -e 's|{ARG_OS}|$(OS)|g' \
 	    -e 's|{ARG_FROM}|$(BASEIMAGE)|g' \
 	    Dockerfile.in > .dockerfile-$(OS)_$(ARCH)
-	@docker build -t $(IMAGE):$(TAG) -f .dockerfile-$(OS)_$(ARCH) .
+	@docker build --build-arg HTTP_PROXY=$(HTTP_PROXY) --build-arg=$(HTTPS_PROXY) -t $(IMAGE):$(TAG) -f .dockerfile-$(OS)_$(ARCH) .
```
Comment (thockin) [inline Makefile]: Can we line-wrap this as

```
@docker build \
    --build-arg HTTP_PROXY=$(HTTP_PROXY) \
    --build-arg=$(HTTPS_PROXY) \
    -t $(IMAGE):$(TAG) \
    -f .dockerfile-$(OS)_$(ARCH) \
    .
````

File: Makefile [hunk 4]
```diff
@@ -155,6 +156,8 @@ test: build-dirs
 	    -v $$(pwd)/bin/$(OS)_$(ARCH):/go/bin                                     \
 	    -v $$(pwd)/.go/std/$(OS)_$(ARCH):/usr/local/go/pkg/$(OS)_$(ARCH)_static  \
 	    -v $$(pwd)/.go/cache:/.cache                                             \
+            --env HTTP_PROXY=$(HTTP_PROXY)                                 	     \
```
Comment (thockin) [inline Makefile]: same whitespace comment as above please

File: Makefile (hunk 3 above)
Comment (thockin) [reply Makefile]: Do we really need the build-arg?  The env var will be passed in and used in the build lines.

Comment (chris060986) [reply Makefile]: Yes, I double checked behind the proxy. If i'm not passing the proxy to the docker build Step 4 will fail. See logs:
`building: bin/linux_amd64/git-sync
Sending build context to Docker daemon  13.39MB
Step 1/7 : FROM alpine:3.8
 ---> 196d12cf6ab1
Step 2/7 : MAINTAINER Tim Hockin <thockin@google.com>
 ---> Using cache
 ---> 452cffc36068
Step 3/7 : ADD bin/linux_amd64/git-sync /git-sync
 ---> f7b715a845a7
Step 4/7 : RUN apk update --no-cache && apk add     ca-certificates     coreutils     git     openssh-client
 ---> Running in cacefe191798
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
WARNING: Ignoring http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz: temporary error (try again later)
OK: 13 distinct packages available
WARNING: Ignoring http://dl-cdn.alpinelinux.org/alpine/v3.8/community/x86_64/APKINDEX.tar.gz: temporary error (try again later)
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.8/community/x86_64/APKINDEX.tar.gz
ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.8/main: temporary error (try again later)
WARNING: Ignoring APKINDEX.adfa7ceb.tar.gz: No such file or directory
ERROR: http://dl-cdn.alpinelinux.org/alpine/v3.8/community: temporary error (try again later)
WARNING: Ignoring APKINDEX.efaa1f73.tar.gz: No such file or directory
ERROR: unsatisfiable constraints:
  ca-certificates (missing):
    required by: world[ca-certificates]
  coreutils (missing):
    required by: world[coreutils]
  git (missing):
    required by: world[git]
  openssh-client (missing):
    required by: world[openssh-client]`

**Comment by chris060986** [comment]
@thockin I try to fix the intention with VS Code. If you still have issues please let me know which editor you are using. Thanks!

File: Makefile [hunk 5]
```diff
@@ -119,7 +121,12 @@ container: .container-$(DOTFILE_IMAGE) container-name
 	    -e 's|{ARG_OS}|$(OS)|g' \
 	    -e 's|{ARG_FROM}|$(BASEIMAGE)|g' \
 	    Dockerfile.in > .dockerfile-$(OS)_$(ARCH)
-	@docker build -t $(IMAGE):$(TAG) -f .dockerfile-$(OS)_$(ARCH) .
+	@docker build \
+	    --build-arg HTTP_PROXY=$(HTTP_PROXY) \
```
Comment (thockin) [inline Makefile]: These 2 lines are very different - on purpose?  It doesn't seem to be..

>	    --build-arg HTTP_PROXY=$(HTTP_PROXY) \

defines KEY=VAL

>	    --build-arg=$(HTTPS_PROXY) \

defines just KEY

**Comment by thockin** [comment]
Thanks!!


/lgtm
/approve

**Pull request merged**